tests/server/testServerWithBaseTheme.js

```diff
@@ -0,0 +1,109 @@
+            type: "kettle.test.request.http",
+            options: {
+                path: "/clientConfig",
+                method: "GET"
+            }
+        },
+        baseThemeFileBothExistRequest: {
+            type: "kettle.test.request.http",
+            options: {
+                path: "/testFile.txt",
+                method: "GET"
+            }
+        },
+        baseThemeFileBaseOnlyRequest: {
+            type: "kettle.test.request.http",
+            options: {
+                path: "/testBaseFile.txt",
+                method: "GET"
+            }
+        },
+        baseThemeFileCustomOnlyRequest: {
+            type: "kettle.test.request.http",
+            options: {
+                path: "/testCustomFile.txt",
+                method: "GET"
+            }
+        },
+        baseThemeFileMissingRequest: {
+            type: "kettle.test.request.http",
+            options: {
+                path: "/notARealFile.txt",
+                method: "GET"
+            }
+        }
+    },
+    sequence: [{
+        // test the clientConfig info endpoint
+        func: "{clientConfigRequest}.send"
+    }, {
+        event: "{clientConfigRequest}.events.onComplete",
+        listener: "sjrk.storyTelling.server.verifyGetClientConfigSuccessful",
+        args: ["{arguments}.0", "{arguments}.1", undefined]
+    }, {
+        // test getting a file that exists for both base and custom theme, with no custom theme specified
+        func: "{baseThemeFileBothExistRequest}.send"
+    }, {
+        event: "{baseThemeFileBothExistRequest}.events.onComplete",
+        listener: "sjrk.storyTelling.server.verifyGetThemeFileSuccessful",
+        args: ["{arguments}.0", "{arguments}.1", "This is the base test file"]
+    }, {
+        // test getting a file that exists only for base theme, with no custom theme specified
+        func: "{baseThemeFileBaseOnlyRequest}.send"
+    }, {
+        event: "{baseThemeFileBaseOnlyRequest}.events.onComplete",
+        listener: "sjrk.storyTelling.server.verifyGetThemeFileSuccessful",
+        args: ["{arguments}.0", "{arguments}.1", "Test file that isn't in the custom theme"]
+    }, {
+        // test getting a file that exists only for custom theme, with no custom theme specified
+        func: "{baseThemeFileCustomOnlyRequest}.send"
+    }, {
+        event: "{baseThemeFileCustomOnlyRequest}.events.onComplete",
+        listener: "sjrk.storyTelling.server.verifyGetThemeFileUnsuccessful",
+        args: ["{arguments}.0", "{arguments}.1"]
+    }, {
+        // test getting a file that exists only for custom theme, with no custom theme specified
+        func: "{baseThemeFileMissingRequest}.send"
+    }, {
+        event: "{baseThemeFileMissingRequest}.events.onComplete",
+        listener: "sjrk.storyTelling.server.verifyGetThemeFileUnsuccessful",
+        args: ["{arguments}.0", "{arguments}.1"]
+    }]
+}];
+
+kettle.test.bootstrapServer(sjrk.storyTelling.server.testServerWithBaseThemeDefs);
```

Comment:
We should test rendering of actual theme UIs. Is it where you have issues with UIO - https://issues.fluidproject.org/browse/SJRK-255?

Reply:
Yes, the tests I was trying to add when I ran into the SJRK-255 issue were to add calls to `sjrk.storyTelling.loadStoryFromParameter()` and `sjrk.storyTelling.loadBrowse()`, which would each load in actual themed UIs.

Mysteriously, while working on SJRK-255 today, I found myself unable to reproduce the error.